### DISCUSSION TITLE
srsRAN gNB with srsUE: ZeroMQ-based Setup - Error getting output from iperf3

### DISCUSSION THREAD

**Discussion by 4NE3Q** [srsran/srsRAN_Project#434] [General Help] [answered]
Hi,

I have setup the whole thing ZeroMQ-based and everything works fine but when I try to do iperf3 the very last step I get this error:

ip netns exec ue1 iperf3 -c 10.45.1.1 -b 10M -i 1 -t 60 -u
iperf3: error - unable to send control message: Bad file descriptor

I really need help regarding this and I have tried searching for this and I'm unable to find any solutions to this. Any help will be appreciated. Thanks

**Comment by initChat** [comment]
Can you elaborate more of your setting. And does the gNB and open5gs returns any result when the iperf shows error.

**Comment by pgawlowicz** [comment]
@4NE3Q iperf3 requires server to be running also for the UDP connection. Do you run the iperf3 server on the Open5gc side? You can use just `iperf` which does not require the server for UDP, and just sends traffic.

**Comment by mas-oliveira** [comment]
Check if you have created the Namespace to ue1 during that session, sometimes you restart the machine and maybe you forgot to:
sudo ip netns add ue1

**Reply by pgawlowicz** [reply]
@4NE3Q did you fix your issue?

**Comment by pgawlowicz** [comment, marked as answer]
@4NE3Q Indeed there was a bug in the tutorial. Please check the updated version:
https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#id13

You can also find alternative solutions here: https://github.com/srsran/srsRAN_Project/issues/327#issuecomment-1923331500
